### PR TITLE
QA pipeline BART compatible

### PR DESCRIPTION
**Pull request by mfuntowicz**

### CODE REVIEW

File: src/transformers/pipelines.py [hunk 1]
```diff
@@ -1293,6 +1296,10 @@ def __call__(self, *args, **kwargs):
                 # Mask CLS
                 start_[0] = end_[0] = 0
 
+                # Make masking the lowest value around there
```
Comment (patrickvonplaten) [inline src/transformers/pipelines.py:1299]: hard to understand the comment here

File: src/transformers/pipelines.py [hunk 2]
```diff
@@ -1284,6 +1284,9 @@ def __call__(self, *args, **kwargs):
             min_null_score = 1000000  # large and positive
             answers = []
             for (feature, start_, end_) in zip(features, start, end):
+                # Ensure padding cannot be attended
+                start_, end_ = start_ * feature.attention_mask, end_ * feature.attention_mask
```
Comment (patrickvonplaten) [inline src/transformers/pipelines.py:1288]: nit => maybe in two lines for better readability

File: src/transformers/pipelines.py [hunk 3]
```diff
@@ -1284,6 +1284,9 @@ def __call__(self, *args, **kwargs):
             min_null_score = 1000000  # large and positive
             answers = []
             for (feature, start_, end_) in zip(features, start, end):
+                # Ensure padding cannot be attended
```
Comment (patrickvonplaten) [inline src/transformers/pipelines.py:1287]: Maybe improve comment to => "cannot be attended" makes me think of preparing inputs to a model to not be attended by the self-attention layers, but `start` and `end` are the outputs of the model.

Do we even need the lines `start_, end_ = start_ * feature.attention_mask, end_ * feature.attention_mask`? 
Can't we just set all index whose `attention_mask` is 0 to -10000 later?

Comment (patrickvonplaten) [reply src/transformers/pipelines.py:1287]: Otherwise maybe just change comment to `Zero out start and end logits at masked positions`

**Review by patrickvonplaten — APPROVED** [review]
Nice!

**Review by LysandreJik — APPROVED** [review]
Very nice!

File: src/transformers/data/processors/squad.py [hunk 4]
```diff
@@ -123,9 +123,11 @@ def squad_convert_example_to_features(example, max_seq_length, doc_stride, max_q
     truncated_query = tokenizer.encode(
         example.question_text, add_special_tokens=False, truncation=True, max_length=max_query_length
     )
+
+    tokenizer_type = type(tokenizer).__name__.replace("Tokenizer", "").lower()
```
Comment (sshleifer) [inline src/transformers/data/processors/squad.py:127]: comment here would help! May also apply to GPT2 (which roberta and bart inherit from)

**Review by sshleifer — APPROVED** [review]
